tests/components/withings/test_config_flow.py

```diff
@@ -1,45 +1,100 @@
 """Tests for config flow."""
+from aiohttp.test_utils import TestClient
 
+from homeassistant.components import webhook
 from homeassistant.components.withings import const
-from homeassistant.components.withings.config_flow import WithingsFlowHandler
-from homeassistant.core import HomeAssistant
+from homeassistant.config import async_process_ha_core_config
+from homeassistant.const import (
+    CONF_CLIENT_ID,
+    CONF_CLIENT_SECRET,
+    CONF_EXTERNAL_URL,
+    CONF_UNIT_SYSTEM,
+    CONF_UNIT_SYSTEM_METRIC,
+)
+from homeassistant.core import DOMAIN as HA_DOMAIN, HomeAssistant
+from homeassistant.helpers import config_entry_oauth2_flow
+from homeassistant.helpers.config_entry_oauth2_flow import AUTH_CALLBACK_PATH
+from homeassistant.setup import async_setup_component
 
-from tests.components.withings.common import ComponentFactory, new_profile_config
+from tests.common import MockConfigEntry
 
 
-async def test_config_non_unique_profile(
-    hass: HomeAssistant, component_factory: ComponentFactory
-) -> None:
+async def test_config_non_unique_profile(hass: HomeAssistant) -> None:
     """Test setup a non-unique profile."""
-    person0 = new_profile_config("person0", 0)
+    config_entry = MockConfigEntry(
+        domain=const.DOMAIN, data={const.PROFILE: "person0"}, unique_id="0"
+    )
+    config_entry.add_to_hass(hass)
 
-    await component_factory.configure_component(profile_configs=(person0,))
-    await component_factory.setup_profile(person0.user_id)
+    result = await hass.config_entries.flow.async_init(
+        const.DOMAIN, context={"source": "profile"}, data={const.PROFILE: "person0"}
+    )
 
-    flow_handler = WithingsFlowHandler()
-    flow_handler.hass = hass
-    flow_handler.context = {}
-    result = await flow_handler.async_oauth_create_entry({"profile": person0.profile})
     assert result
     assert result["errors"]["base"] == "profile_exists"
 
 
 async def test_config_reauth_profile(
-    hass: HomeAssistant, component_factory: ComponentFactory
+    hass: HomeAssistant, aiohttp_client, aioclient_mock
 ) -> None:
     """Test reauth an existing profile re-creates the config entry."""
-    person0 = new_profile_config("person0", 0)
+    hass_config = {
+        HA_DOMAIN: {
+            CONF_UNIT_SYSTEM: CONF_UNIT_SYSTEM_METRIC,
+            CONF_EXTERNAL_URL: "http://127.0.0.1:8080/",
+        },
+        const.DOMAIN: {
+            CONF_CLIENT_ID: "my_client_id",
+            CONF_CLIENT_SECRET: "my_client_secret",
+            const.CONF_USE_WEBHOOK: False,
+        },
+    }
+    await async_process_ha_core_config(hass, hass_config.get(HA_DOMAIN))
+    assert await async_setup_component(hass, HA_DOMAIN, {})
+    assert await async_setup_component(hass, webhook.DOMAIN, hass_config)
```

Comment:
We should only need to set up the withings integration. The other integrations are dependencies of this integration.